src/util.rs

```diff
@@ -0,0 +1,24 @@
+use core::sync::atomic::{AtomicBool, Ordering};
+
+#[derive(Debug)]
+pub struct LazyBool {
+    value: AtomicBool,
+    needs_init: AtomicBool,
```

Comment:
I think using one atomic and bit flags will be more efficient than two `AtomicBool`s. Plus you will be able to relax orderings in `usync_init`.

Reply:
Done, now both `LazyBool` and `Weak` use `LazyUsize`. `LazyUsize` itself just uses an `AtomicUsize` with `Ordering::Relaxed`.